src/components/Table/GridTable.tsx

```diff
@@ -217,6 +221,8 @@ export interface GridTableProps<R extends Kinded, S, X> {
   xss?: X;
   /** Experimental API allowing one to scroll to a table index. Primarily intended for stories at the moment */
   api?: MutableRefObject<GridTableApi | undefined>;
+  /** Specify the element in which the table should resize its columns against. If not set, the table will resize columns based on its owns container's width */
+  resizeTarget?: MutableRefObject<HTMLElement | null>;
```

Comment:
This is required by SchedulesV2. When the Task Details pane opens, then the table container (`tableRef`) will resize. But we don't actually want to base the table width off of this value. Instead we want the Schedule table to stay its same width but introduce a horizontal scrollbar. In this instance the resizeTarget is the container of both the Schedule Table and the Task Details pane.

Reply:
This seems like too much of a hack / non-feature imo, and I've been hoping we could figure out a way to do this without the table being aware of the side panel being open next to it.

I.e. could we accomplish this with some sort of SidePanelContainer component? That manages the left child to always be 100% of the main width, and then when the side panel opens, "do something" to achieve the look & feel w/o whatever content is in the left child (i.e. the table in this instance) being none the wiser?

Even if SidePanelContainer had to have its own resize observer, i.e. it used the same trick of "capture width as a pixel and fix my left child's width to _always_ be that same px width, even after the right panel opens", I think I'd prefer to have the two separate resize observers (one in SidePanelContainer and one in GridTable) b/c they just seem like orthogonal concerns.

If that seems doable, could we leave resizeTarget out of GridTable and let Schedules V2 keep doing it's local hack until we have a SidePanelContainer (or whatever a better name is) solution? Or do we really need to do `resizeTarget` now and we could follow up with removing it when/if we have a dedicated SidePanelContainer solution?

Reply:
(All ^ said, I'm fine merging this as-is to GSD, and keeping it on a todo list to clean up if my nudge towards a SidePanelContainer is actually doable/wouldn't be that bad.)

Reply:
Yea, I haven't thought too much about including a SidePanelContainer directly in GridTable, but definitely something that we might want. The Documents page in Blueprint has a persistent drawer open, which could benefit from this. Though, the big difference between that Documents and the Schedule Table drawers would be the Schedule's table needs to keep its width when the drawer opens, while the Documents table would try and avoid a horizontal scrollbar while the drawer is open. 
I would prefer to keep this as is for now and I'll update the comment to discourage people from using it.

Reply:
> SidePanelContainer directly in GridTable

Just checking, but "in GridTable" --> "in Beam" is what I was thinking, such that GridTable wouldn't have to know whether it is/is not in a side panel-based UX.

> Schedule's table needs to keep its width when the drawer opens, while the
> Documents table would try and avoid a horizontal scrollbar while the drawer is open.

Hopefully we could get UX to just pick one, or yeah we end up with a `SidePanelContainer` that has a flag about which style of UX it uses, and then either way each page's main GridTable is blithely unaware. As the goal anyway.

> I would prefer to keep this as is for now and I'll update the comment
> to discourage people from using it.

Cool, sgtm.

Reply:
Gotcha - I was assuming you were referencing a side panel that GridTable is aware of / the side pane was an option of GridTable. That is how its been poised from Design, at least on the Documents page: "Table with persistent drawer". Though, I agree with you and would prefer to keep them separate.